### ISSUE TITLE
Cannot parse privateKey: Unsupported key format

### ISSUE THREAD

**Issue by doedje** [mscdex/ssh2-streams#100]
I recently updated my RSA public/private key to use the OpenSSH key format, the file now begins with:
```
-----BEGIN OPENSSH PRIVATE KEY-----
```
But while I don't have any problem with other programs, ftp-remote-edit (a package for the Atom text editor: https://github.com/h3imdall/ftp-remote-edit) is the only software that has an issue and is giving me the error:
```
Cannot parse privateKey: Unsupported key format
```
I dived into the matter and found out the error was thrown by ssh2-stream's keyParser module. Rather logical when reading the script, the header of my keyfile does not match any of the regular expressions.

To convert my key to the OpenSSH format I used:
```
ssh-keygen -f id_rsa -o -p -a 100
```
Converting it back to the old format can be done with:
```
ssh-keygen -f id_rsa -p -a 100
```
Which does indeed solve the issue, but the only problem is this will not work for my ED25519 keypairs since they only accept OpenSSH formatting.

**Comment by mscdex** [comment]
FWIW Ed25519 keys are not supported because as of this writing there is no version of node bundled with a version of OpenSSL that supports such keys.

**Comment by doedje** [comment]
I was rather surprised (and confused) because also my RSA key in openSSH format (created with `-o`) was not accepted while the documentation of ssh2 states it is compatible with openSSH formatted keys....

But I will investigate the issue further down the rabbit hole. :)

**Comment by mscdex** [comment]
> while the documentation of ssh2 states it is compatible with openSSH formatted keys....

The `ssh2` module pre-dates `ssh-keygen`'s `-o` option, so that is mostly the reason why. At that time the "OpenSSH format" was the more general PEM format. I have no problem adding support for the new format though.

**Comment by mscdex** [comment]
Full Ed25519 support now available in `ssh2-streams` v0.4.3.
